src/test/java/TypewiseAlert/TypewiseAlertTest.java

```diff
@@ -10,6 +10,49 @@ public class TypewiseAlertTest
     public void infersBreachAsPerLimits()
     {
       assertTrue(TypewiseAlert.inferBreach(12, 20, 30) ==
-        TypewiseAlert.BreachType.TOO_LOW);
+        BreachTypes.TOO_LOW);
+    }
+    
+    @Test
+    public void infersBreachAsPerHighLimits()
+    {
+      assertTrue(TypewiseAlert.inferBreach(35, 20, 30) ==
+    		  BreachTypes.TOO_HIGH);
+    }
+    
+    @Test
+    public void infersBreachAsPerNormalLimits()
+    {
+      assertTrue(TypewiseAlert.inferBreach(25, 20, 30) ==
+    		  BreachTypes.NORMAL);
+    }
+    
+    //================================================
+    
+    @Test
+    public void checkAlert_controller_Passive_cooling() {
+        assertTrue(TypewiseAlert.checkAndAlert(AlertTargets.TO_CONTROLLER,CoolingTypes.PASSIVE_COOLING,31));
```

Comment:
there are totally 2 targets, 3 cooling types , and 5 values (2 boundary , 1 low, 1 normal, 1 very high), which means there could be potentially 30 variations. Please check if important variations are considered.